stdlib/private/SwiftPrivateLibcExtras/Subprocess.swift

```diff
@@ -64,12 +67,74 @@ func posixPipe() -> (readFD: CInt, writeFD: CInt) {
 /// stderr.
 public func spawnChild(_ args: [String])
   -> (pid: pid_t, stdinFD: CInt, stdoutFD: CInt, stderrFD: CInt) {
+  // The stdout, stdin, and stderr from the child process will be redirected
+  // to these pipes.
+  let childStdout = posixPipe()
+  let childStdin = posixPipe()
+  let childStderr = posixPipe()
+
+#if os(Android)
+  // posix_spawn isn't available on Android. Instead, we fork and exec.
+  // To correctly communicate the exit status of the child process to this
+  // (parent) process, we'll use this pipe.
+  let childToParentPipe = posixPipe()
+
+  let pid = fork()
+  precondition(pid >= 0, "fork() failed")
+  if pid == 0 {
+    // pid of 0 means we are now in the child process.
+    // Capture the output before executing the program.
+    dup2(childStdout.writeFD, STDOUT_FILENO)
+    dup2(childStdin.readFD, STDIN_FILENO)
+    dup2(childStderr.writeFD, STDERR_FILENO)
+
+    // Set the "close on exec" flag on the parent write pipe. This will
+    // close the pipe if the execve() below successfully executes a child
+    // process.
+    let closeResult = fcntl(childToParentPipe.writeFD, F_SETFD, FD_CLOEXEC)
+    let closeErrno = errno
+    precondition(
+      closeResult == 0,
+      "Could not set the close behavior of the child-to-parent pipe; " +
+      "errno: \(closeErrno)")
+
+    // Start the executable. If execve() does not encounter an error, the
+    // code after this block will never be executed, and the parent write pipe
+    // will be closed.
+    withArrayOfCStrings([Process.arguments[0]] + args) {
+      execve(Process.arguments[0], $0, _getEnviron())
+    }
+
+    // If execve() encountered an error, we write the errno encountered to the
+    // parent write pipe.
+    let errnoSize = sizeof(errno.dynamicType)
+    var execveErrno = errno
+    let writtenBytes = withUnsafePointer(&execveErrno) {
+      write(childToParentPipe.writeFD, UnsafePointer($0), errnoSize)
+    }
+
+    let writeErrno = errno
+    if writtenBytes > 0 && writtenBytes < errnoSize {
+      // We were able to write some of our error, but not all of it.
+      // FIXME: Retry in this case.
+      preconditionFailure("Unable to write entire error to child-to-parent " +
+                          "pipe.")
+    } else if writtenBytes == 0 {
+      preconditionFailure("Unable to write error to child-to-parent pipe.")
+    } else if writtenBytes < 0 {
+      preconditionFailure("An error occurred when writing error to " +
+                          "child-to-parent pipe; errno: \(writeErrno)")
+    }
+
+    // Close the pipe when we're done writing the error.
+    close(childToParentPipe.writeFD)
```

Comment:
Still stumped by this. [This is what I have locally](https://gist.github.com/modocache/5b185c5ec0b487fa92067c2019da6d8f), but the test suite stalls out at `test/1_stdlib/Map.swift` -- that test never finishes executing.

@gribozavr I'm tempted to ask whether we can punt this work to #1714, since the tests aren't even capable of being run for Android with just the changes in this pull request. At this point, anything that compiles would theoretically be acceptable.